src/bus/mod.rs

```diff
@@ -164,8 +164,19 @@ impl Bus {
 
                 addr
             }
-            UnixSocket::Dir(_) => bail!("`dir` transport is not supported (yet)."),
-            UnixSocket::TmpDir(_) => bail!("`tmpdir` transport is not supported (yet)."),
+            UnixSocket::Dir(dir) | UnixSocket::TmpDir(dir) => {
+                // for tmpdir, the server _may_ create an abstract socket but isn't required to do so
+                // https://dbus.freedesktop.org/doc/dbus-specification.html
+                // so our tmpdir is the same as dir, as the specification suggests
+                let path = dir.join(format!("dbus-{}", rand::random::<u32>()));
```

Comment:
I'm not sure if we need to pull in the big guns here. We already depend indirectly on `fastrand` and that should suffice?